### ISSUE TITLE
got bad performance on bert pre-train

### ISSUE THREAD

**Issue by 652994331** [NVIDIA/DeepLearningExamples#184]
Hi,guys, might need some help here. it seems the create_pretrain_data has no differece from the bert's official create_pretrain_data.py. So I use some domain data to generate tf.records. i have 4 gpus so i first separated the data into 4 different parts. then i use create_pretrain_data.py 4 times to generate 4 different tf.records. run the run_pretrain.sh with horovod, the lose was large at first like 4, it reduced to around 1.5 soon, but no less. After 38000steps， the loss became worse, around 6. the performance of the pretrained bert is very bad. Anyone has idea? thanks

**Comment by swethmandava** [comment]
We do observe loss stagnating in our experiments. Downstream tasks, however, get better with more iterations.

With regards to loss becoming worse, that indicates the model diverging. You might want to decrease the learning rate and/or increase warmup

**Comment by 652994331** [comment]
@swethmandava Thanks, for now i am think to set learning rate to 1e-5. and wait longer, maybe at a larger step like 100k, the loss will reduce again.

**Comment by sharathts** [comment]
Loss spiking at any iteration is not a good sign. It might go down later on, but is indicative of an aggressive learning rate.

**Comment by 652994331** [comment]
@sharathts thanks for your reply, what do u mean an aggressive learing rate? i am sorry i am new here.

**Comment by sharathts** [comment]
Less aggressive is to use a smaller learning rate.
